onnxruntime/core/providers/qnn/qnn_telemetry.h

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#ifdef _WIN32
+#include <Windows.h>
+
+#if !BUILD_QNN_EP_STATIC_LIB
+#include <TraceLoggingProvider.h>
+#endif
+
+#include <functional>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "core/providers/qnn/ort_api.h"
+
+#if !BUILD_QNN_EP_STATIC_LIB
+TRACELOGGING_DECLARE_PROVIDER(telemetry_provider_handle);
+#endif
+
+namespace onnxruntime {
+namespace qnn {
+
+/// <summary>
+/// Singleton class used to log QNN profiling events to the ONNX Runtime telemetry tracelogging provider.
+///
+/// When QNN EP is a DLL, we must define our own tracelogging provider handle via TRACELOGGING_DEFINE_PROVIDER.
+/// TraceLogging documentation states that separate DLLs cannot share the same tracelogging provider handle. See:
+/// https://learn.microsoft.com/en-us/windows/win32/api/traceloggingprovider/nf-traceloggingprovider-tracelogging_define_provider#remarks
+///
+/// When QNN EP is a static library, we use the tracelogging provider handle already defined
+/// in core/platform/windows/telemetry.h/.cc. In this case, we forward method calls to the
+/// ORT Env's telemetry provider.
+/// </summary>
+class QnnTelemetry {
```

Comment:
Hi @ivberb, If you're available, I'd would really appreciate a review of the ETW changes in this PR. Thanks.

Reply:
ETW changes look fine. Thanks @adrianlizarraga !